frontend/lib/src/components/widgets/DataFrame/DataFrame.tsx

```diff
@@ -262,6 +234,86 @@ function DataFrame({
   const { columns, sortColumn, getOriginalIndex, getCellContent } =
     useColumnSort(originalNumRows, originalColumns, getOriginalCellContent)
 
+  const applySelections = React.useCallback(
+    // Use debounce to prevent rapid updates to the widget state.
+    debounce(DEBOUNCE_TIME_MS, (newSelection: GridSelection) => {
+      // If we want to support selections also with in the editable mode,
+      // we would need to integrate the `applyEdits` and `applySelections` functions
+      // into a single function that updates the widget state with both the editing
+      // state and the selection state.
+
+      const selectionState = {
+        // We use snake case here since this is the widget state
+        // that is sent and used in the backend. Therefore, it should
+        // conform with the Python naming conventions.
```

Comment:
nit: the fields look like snake_case & camelcase would be the same for them since they are single words. Maybe if would make sense to rewrite it slightly to make this clear: 
`If you intend to rename the columns, remember to use snake_case since ...`